nifi-nar-bundles/nifi-workday-bundle/nifi-workday-processors/src/main/java/org/apache/nifi/processors/workday/GetWorkdayReport.java

```diff
@@ -0,0 +1,466 @@
+        FlowFile responseFlowFile = null;
+        String hashingAlgorithm = context.getProperty(HASHING_ALGORITHM).getValue();
+        Set<String> columnsToHash = Optional.ofNullable(
+            context.getProperty(FIELDS_TO_HASH)
+                .evaluateAttributeExpressions(flowfile)
+                .getValue())
+            .map(String::trim)
+            .map(columns -> columns.split(COLUMNS_TO_HASH_SEPARATOR))
+            .map(Arrays::stream)
+            .map(columns -> columns.collect(Collectors.toSet()))
+            .orElse(Collections.emptySet());
+        try {
+            if (isSuccess(httpResponseEntity.statusCode())) {
+                responseFlowFile = flowfile == null ? session.create() : session.create(flowfile);
+                InputStream responseBodyStream = httpResponseEntity.body();
+                if (recordReaderFactoryReference.get() != null) {
+                    TransformResult transformResult = transformRecords(session, flowfile, responseFlowFile, hashingAlgorithm, columnsToHash, responseBodyStream);
+                    Map<String, String> attributes = new HashMap<>();
+                    attributes.put(RECORD_COUNT, String.valueOf(transformResult.getNumberOfRecords()));
+                    attributes.put(CoreAttributes.MIME_TYPE.key(), transformResult.getMimeType());
+                    responseFlowFile = session.putAllAttributes(responseFlowFile, attributes);
+                } else {
+                    responseFlowFile = session.importFrom(responseBodyStream, responseFlowFile);
+                    Optional<String> mimeType = httpResponseEntity.headers().getFirstHeader(HEADER_CONTENT_TYPE);
+                    if (mimeType.isPresent()) {
+                        responseFlowFile = session.putAttribute(responseFlowFile, CoreAttributes.MIME_TYPE.key(), mimeType.get());
+                    }
+                }
+            }
+        } catch (Exception e) {
+            session.remove(responseFlowFile);
+            throw e;
+        }
+        return responseFlowFile;
+    }
+
+    private String createAuthorizationHeader(ProcessContext context, FlowFile flowfile) {
+        String userName = context.getProperty(WORKDAY_USERNAME).evaluateAttributeExpressions(flowfile).getValue();
+        String password = context.getProperty(WORKDAY_PASSWORD).evaluateAttributeExpressions(flowfile).getValue();
+        String base64Credential = Base64.getEncoder().encodeToString((userName + USERNAME_PASSWORD_SEPARATOR + password).getBytes(StandardCharsets.UTF_8));
+        return BASIC_PREFIX + base64Credential;
+    }
+
+    private TransformResult transformRecords(ProcessSession session, FlowFile flowfile, FlowFile responseFlowFile, String hashingAlgorithm, Set<String> columnsToHash,
+        InputStream responseBodyStream) throws IOException, SchemaNotFoundException, MalformedRecordException {
+        int numberOfRecords = 0;
+        String mimeType = null;
```

Comment:
The `null` initialized value is not necessary:
```suggestion
        String mimeType;
```

Reply:
done